xarray/core/variable.py

```diff
@@ -1789,12 +1789,17 @@ class IndexVariable(Variable):
     unless another name is given.
     """
 
-    def __init__(self, dims, data, attrs=None, encoding=None, fastpath=False):
+    def __init__(self, dims, data, attrs=None, encoding=None, fastpath=False, name=None):
```

Comment:
I would rather not add an explicit `name` parameter to `IndexVariable`.

If anything, I would think about removing the `name` property. Long term, I'd like to rename `IndexVariable` -> `ImmutableVariable` and move the pandas adapter logic into class that we can pass into the `data` argument of `Variable`.